### DISCUSSION TITLE
Flutter does not build on M2

### DISCUSSION THREAD

**Discussion by Chrysofin** [codemagic-ci-cd/codemagic-docs#2771] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

A build was triggered as per normal and it failed giving the an error. I have tried various things but instead if fixes one and opens up more. 

This builds on a local machine with M2 so not sure what is going on. 

Your help will be greatly appreciated.

Build ID's affected. 

6704590874ee965d3c15d94d
670375889a936e61e74094b2
67045290252925962074141d
6702de907933563aa28782d3-- because of the error in this build we have to use 3.19.6

### Expected results

Should build and deploy to the appstore.

### Actual results

Xcode archive done.                                         13.9s
Failed to build iOS app
Swift Compiler Error (Xcode): Type 'UIApplication' does not conform to protocol 'Launcher'
/Users/builder/.pub-cache/hosted/pub.dev/url_launcher_ios-6.2.5/ios/Classes/Launcher.swift:21:0

Uncategorized (Xcode): Command SwiftCompile failed with a nonzero exit code

Encountered error while archiving for device.

Build failed :|
Failed to build for iOS

### Build id (optional)

6704590874ee965d3c15d94d

**Comment by dam-ease** [comment, marked as answer]
Hello @Chrysofin,
This issue has been fixed in `url_launcher_ios`: 6.3.1. Please see https://github.com/flutter/flutter/issues/155306 and try https://github.com/flutter/flutter/issues/155306#issuecomment-2361006384 to see if it helps. You don't have to use a lower version of flutter, please try this and share how it goes on your end.
